### ISSUE TITLE
Dispatch Load-Event 

### ISSUE THREAD

**Issue by AndreasFaust** [rayriffy/urami#3]
The `Image`-component should dispatch a `load`-event.
This is how I would implement this:

```svelte
<script>
	import { createEventDispatcher } from "svelte";
	const dispatch = createEventDispatcher();

	let img: HTMLImageElement | undefined;
	let loaded = false;

	$: if (img?.complete) {
                 if (loaded) return;
		loaded = true;
		dispatch("load");
	}
</script>
<img
	bind:this={img}
	on:load={() => {
		console.log("loaded");
                 if (loaded) return;
		loaded = true;
		dispatch("load");
	}}
        src={builtSrc}
        srcset={builtSrcset}
        decoding="async"
        loading="lazy"
        {...{
          alt,
          height,
          width,
          class: klass,
        }}
/>
```
In my experience `on:load` does not always work (for example when the image is already in browser-cache), so it is necessary to also check `img.complete`. It works, but maybe there is a more elegant way.

**Comment by rayriffy** [comment]
Hi, I would like to revisit this issue again. Is this behavior still relevant?

**Comment by Silvallis** [comment]
Hi I am not the OP but I think this could be usefull. For my use case I would need a way to show a placeholder until the image is loaded.

**Comment by rayriffy** [comment]
solved at [`@urami/svelte@1.1.0`](https://www.npmjs.com/package/@urami/svelte/v/1.1.0)

**Comment by tguelcan** [comment]
Is there also a possibility to trigger an event in case of an error event? For example, to set a placeholder if a removed image no longer exists. Maybe over the [onerror](https://www.w3schools.com/tags/ev_onerror.asp ) function? Has anyone encountered the same problem?
